sklearn/_build_utils/__init__.py

```diff
@@ -33,3 +35,42 @@ def atlas_not_found(blas_info_):
         cblas_libs = blas_info.pop('libraries', [])
 
     return cblas_libs, blas_info
+
+
+def get_cython_source(filename):
+    is_c_filename = filename.endswith('.c')
+    is_cpp_filename = filename.endswith('.cpp')
+
+    # files in src are .c and .cpp files that are not cython-generated
+    if ('src' + os.sep) in filename and (is_c_filename or is_cpp_filename):
```

Comment:
Or we could add `.pyx` extension when it's a cython file, and add a function like:

```py
import os.path

def no_cythonize(extensions, **_ignore):
    for extension in extensions:
        sources = []
        for sfile in extension.sources:
            path, ext = os.path.splitext(sfile)
            if ext in ('.pyx', '.py'):
                if extension.language == 'c++':
                    ext = '.cpp'
                else:
                    ext = '.c'
                sfile = path + ext
            sources.append(sfile)
        extension.sources[:] = sources
    return extensions
```
(copy-paste from [cython doc](http://cython.readthedocs.io/en/latest/src/reference/compilation.html))